osh/cmd_parse.py

```diff
@@ -1606,18 +1567,19 @@ def ParseYshCase(self, case_kw):
         ate = self._Eat(Id.Lit_LBrace)
         arms_start = word_.BraceToken(ate)
 
-        discriminant = self._NewlineOkForYshCase()
+        discriminant = self.w_parser.NewlineOkForYshCase()
 
         # Note: for now, zero arms are accepted, just like POSIX case $x in esac
         arms = []  # type: List[CaseArm]
         while discriminant != Id.Op_RBrace:
             arm = self.ParseYshCaseArm(discriminant)
             arms.append(arm)
 
-            discriminant = self._NewlineOkForYshCase()
+            discriminant = self.w_parser.NewlineOkForYshCase()
 
-        ate = self._Eat(Id.Lit_RBrace)
-        arms_end = word_.BraceToken(ate)
+        ate = self._Eat(Id.Op_RBrace)
```

Comment:
This is a bit subtle -- I guess it means we're always in lex_mode_e.Expr  coming out of NewlineOkForYshCase()

And that sort of "infects" the WordParser, and we have to change it the "normal" Lit_LBrace

The difference is that in `echo }` it's Kind.Lit,  while in ` var d = {}` it's Kind.Op

Maybe add a comment about this